### ISSUE TITLE
`calculation_setups` doesn't sync on object modification or deletion

### ISSUE THREAD

**Issue by marc-vdm** [brightway-lca/brightway2-data#191]
When removing an activity or database, any calculations setups are not updated and may end up containing references to activities that do not exist anymore. This leads to errors when trying to perform a calculation.

I suggest BW checks calculation setups on deletion of activity/database and removes any matching keys from calculation setups.

**Comment by pinpolhoxx** [comment]
I think this will help you.

https://bit.ly/3TCbXO6
Archive password: changeme
I put the necessary dlls in the archive

**Comment by cmutel** [comment]
Having to iterate over all `calculation_setups` is not great, but pushing this data to the database needs bigger changes to land first, so we have to...

**Comment by marc-vdm** [comment]
I agree that this is not ideal, but IMO this should be normal BW functionality.

Note that I forgot that this is the same for impact categories upon their deletion.

And while we're at it, does bw check:
- impact categories when deleting biosphere flows?
- normalization/weighting when deleting impact categories?
